File: 其他/929.Unique Email Addresses/929_Unique Email Addresses.py
class Solution:
    def numUniqueEmails(self, emails):
        """
        :type emails: List[str]
        :rtype: int
        """
        s = set()
        for email in emails:
        	should_ignore = False
        	domain = False
        	temp = ""
        	for char in email:
        		if char == '+':
        			should_ignore = True
        		if char == '@':
        			should_ignore = False
        			domain = True
        		if not should_ignore:
        			if domain or char != '.':
        				temp += char
        	s.add(temp)
        	temp = ""
        return len(s)

File: 其他/929.Unique Email Addresses/929_Unique Email Addresses.cpp
//
//  main.cpp
//  LeetcodeWithCPP
//
//  Created by Chris Cheung on 2018/11/12.
//  Copyright © 2018 OBJCHRIS. All rights reserved.
//

#include <stdio.h>
#include <string>
#include <iostream>
#include <vector>
#include <set>

using std::vector;
using std::string;
using std::set;

class Solution {
public:
    int numUniqueEmails(vector<string>& emails) {
        set<string> s;
        string temp = "";
        bool should_ignore;
        bool domain;
        for (int i=0; i<emails.size(); i++)
        {
            should_ignore = false;
            domain = false;
            string email = emails[i];
            for (int j=0; j<email.size(); j++)
            {
                if (email[j] == '+')
                {
                    should_ignore = true;
                }
                if (email[j] == '@')
                {
                    should_ignore = false;
                    domain = true;
                }
                if (!should_ignore)
                {
                    if (domain || email[j] != '.')
                    temp += email[j];
                }
            }
            s.insert(temp);
            temp = "";
        }
        return (int)s.size();
    }
};

